specs/datapackages/README.md

```diff
@@ -0,0 +1,130 @@
+* `license` - mapping to the ones available on the CKAN instance. **Q:** What happens if we can not map it?
+* `author` / `contributors` - these refer to the datapackage creators not the data ones (`sources` is used for this) **Q:** Should be bother trying to map to CKAN's `author` / `author_email` or `maintainer` /  `maintainer_email`?
+
+Data Packages can have extra fields not present in the CKAN schema:
+* `sources` (name, web, email)
+* `image`
+* `base`
+* `dataDependencies` (ordering of the hash keys is important)
+* `schemas` 
+* Arbitrary fields added to the Data Package
+
+If we don't store the original Data Package somewhere we'll need to store these as extras. 
+
+As per **resources**,  Data Packages `resources` can point the actual data files using `url` or `path` + `base`, or have the data embedded with a `data` field. (The current spec allows all of them to be present at the same time, which hopefully we can restrict https://github.com/dataprotocols/dataprotocols/issues/223). If Data Packages have `data` embedded we can push it to the DataStore to make it available.
+
+Most fields map to CKAN core ones. The ones that are not defined are:
+
+* `encoding`
+* `hash`
+* `sources`
+* `license`
+
+This is not as problematic as dataset fields, as we can add arbitrary top level fields to resources.
+
+## Importing Data Packages
+
+### Implementation
+
+At a higher level this seems quite straight-forward: a  `datapackage_import` or similar action which inherits the auth from `package_create` and `package_update` (eg you need to belong to an org by default etc). The action will validate the JSON object, transform it to a CKAN dict and create or update the relevant package. A controller will handle requests from the UI (at `/datapackage/import`).
+In terms of UI there would be a dedicated page for importing Data Packages. This will essentially contain just an upload field to upload the `datapackage.json` or a text file to enter the URL to an accessible one. This page would show validation errors and redirect to the dataset page once it's created.
+
+Perhaps to make it more accessible we can link to this page from the dataset creation form (from the top bar or the left hand column).
+
+### Potential issues
+
+Identifiers: Data Packages are solely identified by the `name` field, which according to the Data Package spec:
+
+> [is a] short url-usable (and preferably human-readable) name of the package. This MUST be lower-case and contain only alphanumeric characters along with ., _ or - characters. It will function as a unique identifier and therefore SHOULD be unique in relation to any registry in which this package will be deposited (and preferably globally unique).
+
+We can not assume that these identifiers will be unique and that they will not clash with existing CKAN datasets. We can append stuff to the name but then we need to consider what happens when we reupload the same Data Package (ie to update the dataset).
```

Comment:
That sounds good, although perhaps I'd use the organization name if possible. I'd assume that if two users on the same org upload a `budget-2016` file they would be working on the same one (and if not they should use different names)